apps/dash-label-properties/app.py

```diff
@@ -0,0 +1,456 @@
+import pandas as pd
+
+import PIL
+from skimage import color, img_as_ubyte
+from plotly import colors
+
+import base64
+import xarray as xr
+
+external_stylesheets = [dbc.themes.BOOTSTRAP, "assets/object_properties_style.css"]
+
+app = dash.Dash(__name__, external_stylesheets=external_stylesheets)
+server = app.server
+
+# Read the image that will be segmented
+filename = (
+    "https://upload.wikimedia.org/wikipedia/commons/a/ac/Monocyte_no_vacuoles.JPG"
+)
+img = io.imread(filename, as_gray=True)[:660:2, :800:2]
+label_array = measure.label(img < filters.threshold_otsu(img))
+current_labels = np.unique(label_array)[np.nonzero(np.unique(label_array))]
+# Compute and store properties of the labeled image
+prop_names = [
+    "label",
+    "area",
+    "perimeter",
+    "eccentricity",
+    "euler_number",
+    "mean_intensity",
+]
+prop_table = measure.regionprops_table(
+    label_array, intensity_image=img, properties=prop_names
+)
+table = pd.DataFrame(prop_table)
+# Format the Table columns
+columns = [
+    {"name": label_name, "id": label_name, "selectable": True}
+    if precision is None
+    else {
+        "name": label_name,
+        "id": label_name,
+        "type": "numeric",
+        "format": Format(precision=precision),
+        "selectable": True,
+    }
+    for label_name, precision in zip(prop_names, (None, None, 4, 4, None, 3))
+]
+# Select the columns that are selected when the app starts
+initial_columns = ["label", "area"]
+
+img = img_as_ubyte(color.gray2rgb(img))
+img = PIL.Image.fromarray(img)
+
+
+def image_with_contour(
+    img, active_labels, data_table, active_columns, color_column, mode="lines"
+):
+    """
+    Figure with contour plot of labels superimposed on background image.
+
+    Parameters
+    ----------
+    img : PIL Image object.
+    active_labels : list
+        the currently visible labels in the datatable
+    data_table : pandas.DataFrame
+        the currently visible entries of the datatable
+    active_columns: list
+        the currently selected columns of the datatable
+    color_column: str
+        name of the datatable column that is used to define the image colorscale
+    """
+    # Filter the label array based on the currently visible labels
+    mask = np.in1d(label_array.ravel(), active_labels, invert=False).reshape(
+        label_array.shape
+    )
+    masked_labels = label_array * mask
+    values = list(table[color_column].values)
+    value_array = np.array([0] + values)[masked_labels]
+    # Prefer a masked array so we can handle integer values
+    # Background cannot be zero when using datashader (because 'span' is not correctly implemented
+    # in current old version)
+    masked_values = np.ma.masked_where(~mask, value_array)
+
+    print("mode is", mode)
+    opacity = 0.4 if mode is None else 1
+
+    fig = px.imshow(img, binary_string=True, binary_backend="jpg",)
+    # Overlay the current labels
+    _, hex_list = zip(*colors.PLOTLY_SCALES["Viridis"])
```

Comment:
can you wrap this part (the shading) in a helper function? It will be clearer and it's also something we could reuse in other apps so it will be easier to find it back.

Reply:
I think we no longer need this code/function in the label-properties app. Do you think we should still wrap it into a helper function and if so, what would be a good place to put it?

Reply:
please remove any code that is not needed any more. In order to keep track of what you did here maybe we could add an example to the imshow page of the plotly doc (on how to pass grayscale image as a binary image string but still use a colormap) but this is quite a hack so maybe we can just remember that this is possible and we'll find the code in the PR if needed?